src/include/ndpi_typedefs.h

```diff
@@ -1512,6 +1512,10 @@ struct ndpi_flow_struct {
   u_int8_t bfcp_stage:1;
   u_int32_t bfcp_conference_id;
 
+  /* NDPI_PROTOCOL_CNP_IP */
+  u_int8_t cnp_stage:1;
+  u_int32_t cnp_session_id;
```

Comment:
Is it really required to add 8 more bytes to the flow struct?

Reply:
> Is it really required to add 8 more bytes to the flow struct?

It's because of alignment, unfortunately.

Reply:
I think @utoni meant if it is possible to improve the detection without any state... like checking also the version field or waiting for multiple pkts (2-3) all matching the pattern....

Reply:
Since the false positives seems to be only on TCP, could we avoid the state for the time being and process only UDP flow? That should be  a easy patch... 
@0xA50C1A1

Reply:
> Since the false positives seems to be only on TCP, could we avoid the state for the time being and process only UDP flow? That should be a easy patch... @0xA50C1A1

Already done.

Reply:
It doesn't seem so... I mean remove `cnp_stage` and `cnp_session_id` and keep only `NDPI_SELECTION_BITMASK_PROTOCOL_V4_V6_UDP_WITH_PAYLOAD`...

Reply:
> It doesn't seem so... I mean remove `cnp_stage` and `cnp_session_id` and keep only `NDPI_SELECTION_BITMASK_PROTOCOL_V4_V6_UDP_WITH_PAYLOAD`... Does it make sense?

Oh yeah, sorry, I forgot about that.